doc/Contribution.md

```diff
@@ -0,0 +1,106 @@
+```
+  git checkout develop
+```
+2. Fetch the remote changes:
+```
+  git fetch iotagent-node-lib
+```
+3. Merge them:
+```
+  git rebase iotagent-node-lib/develop
+```
+
+Contributions following this guidelines will be added to the `develop` branch, and released in the next version. The
+release process is explaind in the *Releasing* section below.
+
+## Branching model
+There are two special branches in the repository:
+
+* `master`: holds the code for the last stable version of the project. It is only updated when a new version is released,
+and its always updated with the current state of `develop`.
+
+* `develop`: contains the last stable development code. New features and bug fixes are always merged to `develop`.
+
+In order to start developing a new feature or refactoring, a new branch should be created with name `task/<taskName>`.
+This branch must be created from the current version of the `develop` branch. Once the new functionality has been
+completed, a Pull Request will be created from the feature branch to `develop`. Remember to check both the linters
+and the tests before creating the Pull Request.
+
+Bug fixes work the same way as other tasks, with the exception of the branch name, that should be called `bug/<bugName>`.
+
+In order to contribute to the repository, these same scheme should be replicated in the forked repositories, so the
+new features or fixes should all come from the current version of `develop` and end up in `develop` again.
+
+All the `task/*` and `bug/*` branches are temporary, and should be removed once they have been merged.
+
+There is another set of branches called `release/<versionNumber>`, one for each version of the product. This branches
+point to each of the released versions of the project, they are permanent and they are created with each release.
+
+## Changelog
+The project contains a version changelog, called CHANGES_NEXT_RELEASE, that can be found in the root of the project.
+Whenever a new feature or bug fix is going to be merged with `develop`, a new entry should be added to this changelog.
+The new entry should contain the reference number of the issue it is solving (if any).
+
+When a new version is released, the changelog is cleared, and remains fixed in the last commit of that version. The
+content of the changelog is also moved to the release description in the Github release.
+
+## Releasing
+The process of making a release consists of the following steps:
+1. Create a new task branch changing the development version number in the package.json (with a sufix `-next`), to the
+new target version (without any sufix), and PR into `develop`.
+2. Create a tag from the last version of `develop` named with the version number and push it to the repository.
+3. Create the release in Github, from the created tag. In the description, add the contents of the Changelog.
+4. Create a release branch from the last version of `develop` named with the version number.
+5. PR `develop` into `master`.
```

Comment:
It could be a "unsafe" merge... as a "Delete branch (develop)" will appear and many developers may click it in a routine way :)

As alternative, out-of-PR merging develop -> master could be considered (the release.sh script used by CB does that).